src/factories/artifactCluster.ts

```diff
@@ -58,12 +89,18 @@ export async function artifactClusterFactory() {
       renderLabelText(ids.length),
     ])
 
-    await renderBg()
+    const width = artifactPaddingLeft + content.width + artifactPaddingRight
+    const height = content.height + artifactPaddingY * 2
 
-    element.visible = true
+    await Promise.all([
+      renderBg({ width, height }),
+      renderBorder({ item: currentIds, width, height }),
+    ])
```

Comment:
Here we're sending details so the `renderBorder` factory will have the items needed to decide whether to render the selected border or not.